lib/util/git/private-key.ts

```diff
@@ -5,59 +5,132 @@ import upath from 'upath';
 import { PLATFORM_GPG_FAILED } from '../../constants/error-messages';
 import { logger } from '../../logger';
 import { exec } from '../exec';
-import { newlineRegex } from '../regex';
+import type { ExecResult } from '../exec/types';
+import { newlineRegex, regEx } from '../regex';
 import { addSecretForSanitizing } from '../sanitize';
 
-let gitPrivateKey: string | undefined;
-let keyId: string | undefined;
+type PrivateKeyFormat = 'gpg' | 'ssh';
 
-export function setPrivateKey(key: string | undefined): void {
-  if (!is.nonEmptyStringAndNotWhitespace(key)) {
-    return;
+const sshKeyRegex = regEx(
+  /-----BEGIN ([A-Z ]+ )?PRIVATE KEY-----(.|\n)*?-----END ([A-Z]+ )?PRIVATE KEY-----/,
+);
+
+let gitPrivateKey: PrivateKey | undefined;
+
+abstract class PrivateKey {
```

Comment:
please do the class refactoring in a separate PR to be merged before ssh support.

Reply:
Done, see #29875.

Reply:
#29875 has been merged and released.